### ISSUE TITLE
VM connection issues

### ISSUE THREAD

**Issue by k5123** [airspy/firmware#18]
When connecting the AS R2 to a Linux VM (VMware Workstation 14), lsusb produces this output:

> root@ubuntu:/home/test# lsusb
> Bus 001 Device 008: ID 1d50:60a1 OpenMoko, Inc. Airspy
> Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
> Bus 002 Device 003: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
> Bus 002 Device 002: ID 0e0f:0003 VMware, Inc. Virtual Mouse
> Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub

When trying airspy_info, the device is not found:

> root@ubuntu:/home/test# airspy_info 
> airspy_lib_version: 1.0.9
> airspy_open() board 1 failed: AIRSPY_ERROR_NOT_FOUND (-5)

Lastly, lsusb -v produces this: 

> root@ubuntu:/home/test# lsusb -v
> Bus 001 Device 008: ID 1d50:60a1 OpenMoko, Inc. Airspy
> 
> Device Descriptor:
>   bLength                18
>   bDescriptorType         1
>   bcdUSB               2.00
>   bDeviceClass            0 
>   bDeviceSubClass         0 
>   bDeviceProtocol         0 
>   bMaxPacketSize0        64
>   idVendor           0x1d50 OpenMoko, Inc.
>   idProduct          0x60a1 Airspy
>   bcdDevice            1.00
>   iManufacturer           1 (error)
>   iProduct                2 (error)
>   iSerial                 3 (error)
>   bNumConfigurations      1
>   Configuration Descriptor:
>     bLength                 9
>     bDescriptorType         2
>     wTotalLength       0x0020
>     bNumInterfaces          1
>     bConfigurationValue     1
>     iConfiguration          0 
>     bmAttributes         0x80
>       (Bus Powered)
>     MaxPower              500mA
>     Interface Descriptor:
>       bLength                 9
>       bDescriptorType         4
>       bInterfaceNumber        0
>       bAlternateSetting       0
>       bNumEndpoints           2
>       bInterfaceClass       255 Vendor Specific Class
>       bInterfaceSubClass    255 Vendor Specific Subclass
>       bInterfaceProtocol    255 Vendor Specific Protocol
>       iInterface              0 
>       Endpoint Descriptor:
>         bLength                 7
>         bDescriptorType         5
>         bEndpointAddress     0x81  EP 1 IN
>         bmAttributes            2
>           Transfer Type            Bulk
>           Synch Type               None
>           Usage Type               Data
>         wMaxPacketSize     0x0200  1x 512 bytes
>         bInterval               0
>       Endpoint Descriptor:
>         bLength                 7
>         bDescriptorType         5
>         bEndpointAddress     0x02  EP 2 OUT
>         bmAttributes            2
>           Transfer Type            Bulk
>           Synch Type               None
>           Usage Type               Data
>         wMaxPacketSize     0x0200  1x 512 bytes
>         bInterval               0
> can't get device qualifier: Resource temporarily unavailable
> can't get debug descriptor: Resource temporarily unavailable
> cannot read device status, Resource temporarily unavailable (11)
> 

When disconnecting the device from the VM back to Windows, it will also not work in Windows until unplugged and reconnected.

**Comment by k5123** [comment]
Addition: I've managed to get lsusb -v to display the device properties (iManufacturer 1, iProduct 2, iSerial 3) correctly, however after executing airspy_info, the device again doesn't show these properties until reconnection. 

**So at this point i'm no longer sure this is a firmware or a linux library/software issue.**

How i managed to get this working: disconnect/reconnect the device physically. After initial connection, the device may be first active in Windows as "AIRSPY" (device manager). Then connect the device to the VM, execute lsusb -v ("-d 0x1d50:" to select the AirSpy specifically). The output will contain no errors for the device.

Then execute airspy_info or any other airspy_* command, and the output error will appear, after which the device needs to be physically reconnected to work anywhere again.

**Comment by k5123** [comment]
Issue related to VMWare, workaround described here:
https://github.com/airspy/airspyone_host/issues/40 and in the wiki.
TL;DR: Edit .vmx file, add line 
`usb.quirks.device0 = "0x1d50:0x60a1 skip-setconfig"`

Issue solved.
